build/build-g++-1.27.sh

```diff
@@ -0,0 +1,88 @@
+#!/bin/bash
+
+set -ex
+
+ROOT=$(pwd)
+VERSION=$1
+if [[ "$VERSION" != "1.27" ]]; then
+    echo "Wrong version"
+    exit 1
+fi
+
+OUTPUT=/root/g++-${VERSION}.tar.xz
+S3OUTPUT=""
+if echo $2 | grep s3://; then
+    S3OUTPUT=$2
+else
+    OUTPUT=${2-/root/g++-${VERSION}.tar.xz}
+fi
+
+# Workaround for Ubuntu builds
+export LIBRARY_PATH=/usr/lib/x86_64-linux-gnu
+STAGING_DIR=/opt/compiler-explorer/g++-${VERSION}
+
+MAJOR=1
+MAJOR_MINOR=1.27
+
+GCCURL=https://gcc.gnu.org/pub/gcc/old-releases/gcc-${MAJOR}/gcc-${MAJOR_MINOR}.tar.bz2
+URL=https://gcc.gnu.org/pub/gcc/old-releases/gcc-${MAJOR}/g++-${MAJOR_MINOR}.0.tar.bz2
+
+# “unload the tapes”
+curl -L ${GCCURL} | tar jxf -
+curl -L ${URL} | tar jxf -
+
+applyPatches() {
+    local PATCH_DIR=${ROOT}/patches/$2
+    local PATCH=""
+    if [[ -d ${PATCH_DIR} ]]; then
+        echo "Applying patches from ${PATCH_DIR}"
+        pushd $1
+        for PATCH in ${PATCH_DIR}/*; do
+            echo "...${PATCH}"
+            patch -p1 <${PATCH}
+        done
+        popd
+    fi
+}
+
+applyPatches gcc-1.27 gcc${MAJOR_MINOR}
```

Comment:
I guess some "1.27" could be replaced by ${MAJOR_MINOR} as you already have that in other places?